### PR TITLE
AP_Hygrometer: add new driver of hygrometer

### PR DESCRIPTION
**Pull request by gitfishup**
The flight controller receive the hygrometer message from CAN device and flight controller send hygrometer message to the ground station.

background:
CUAV developed a new airspeed sensor that can be heated.The airspeed sensor heats up according to the set temperature, so as to prevent airspeed failure due to ice on the head of the airspeed sensor when the fixed-wing UAV is used in high and cold places.The temperature and humidity sensor is used to detect the temperature and humidity of the pitot tube head. If the measured temperature is lower than the set temperature, the airspeed sensor will enable heating to prevent ice.

1637907979915_0CE47D6B-D58F-44ea-BDE7-F50121475ABE

1637908646536_629D9D29-389A-4b14-B086-50DC2ED267A5

1637908659110_8DCE4D08-65CB-4ee6-AC0B-E9A400387A78

logfiles: plane ,rover, copter , sub 

[log.zip](https://github.com/ArduPilot/ardupilot/files/7606947/log.zip)

parameters for setting:
HYGROx_TYPE: 0 for none , 1 for uavcan. 
Note: HYGRO2_TYPE can be set after HYGRO1_TYPE is set to 1;

previous pr:
https://github.com/ArduPilot/ardupilot/pull/19101 
https://github.com/ArduPilot/ardupilot/pull/19294

### CODE REVIEW

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]
This review isn't exhaustive, but should get the PR further along.

Please do not close an old PR and open a new PR to apply suggested changes. It makes it hard to follow the comments and ensure each one got addressed. And takes a lot longer to review. 

Instead add new commits and squash them as necessary using GIT.  Here is some documentation on how to do that, [Link](https://ardupilot.org/dev/docs/git-interactive-rebase.html).

If you need help with GIT we can provide some assistance.

**Comment by cuhome** [comment]
@tridge Can you help me check it?

**Comment by peterbarker** [comment]
@cuhome @gitfishup ArduPilot now consumes hygrometer off the DroneCAN bus in AP_Airspeed_DroneCAN from dronecan_sensors_hygrometer_Hygrometer .

The data is transmitted to the GCS in ` `HYGROMETER_SENSOR` mavlink message.

I think time has passed by this PR, sorry.  If you did want tosuggest a reformulation of our handling of the existing message (or propose consumption of another), please open a fresh PR against ArduPilot master.

Thanks anyway!
